webrecorder/replayweb.page
# Show page thumbnails in pages list

Resolves https://github.com/webrecorder/replayweb.page/issues/199

## Changes
- Shows thumbnail screenshot of the page if thumbnail exists. If thumbnail doesn't exist, falls back to favicon, with final fallback of an empty space.
- Removes index number from sidebar page list.
- Converts `pageentry` and `pages` to typescript

## Testing
1. Run crawl in browsertrix cloud and download WACZ
2. Load file in local RWP. Verify thumbnails are shown
3. Click page to collapse page list to sidebar. Verify thumbnails are still visible
4. Regression test favicons:
  a. Manually crawl pages with AWP extension
  b. Load file in local RWP. Verify favicons are shown

## Follow-ups
If none of the pages in a list has a thumbnail or favicon, the page list shows an empty column. Could be addressed in a hacky way, but it'd be preferred to add thumbnail metadata to the wacz JSON, so that the `pages` component can hide the column altogether if there are no thumbnails. Alternatively, we could show a nicely designed placeholder image.

## Screenshots
**Home page**:
<img width="1512" alt="Screenshot 2023-08-29 at 10 16 04 AM" src="https://github.com/webrecorder/replayweb.page/assets/4672952/cc9b0302-4d4d-4ba5-a955-659eec19e29a">

**With date query**:
<img width="1512" alt="Screenshot 2023-08-29 at 10 16 16 AM" src="https://github.com/webrecorder/replayweb.page/assets/4672952/f8c21f36-f09a-47d7-9067-125ade066f56">

**Sidebar**:
<img width="450" alt="Screenshot 2023-08-29 at 10 16 37 AM" src="https://github.com/webrecorder/replayweb.page/assets/4672952/79b69114-b39c-4caf-86d3-cb9363d11ad6">

**Sidebar with date query**:
<img width="442" alt="Screenshot 2023-08-29 at 10 17 08 AM" src="https://github.com/webrecorder/replayweb.page/assets/4672952/48026ec4-9094-4f59-86b8-cb29323dd6c7">
